src/main/java/gmbs/domain/Ranking.java

```diff
@@ -0,0 +1,55 @@
+package gmbs.domain;
+
+import java.util.Arrays;
+
+public enum Ranking {
+    FIRST(2000_000_000, 6, false),
+    SECOND(30_000_000, 5, true),
+    THIRD(1_500_000, 5, false),
+    FOURTH(50_000, 4, false),
+    FIFTH(5_000, 3, false),
+    NONE(0, 0, false);
+
+    private final int prize;
+    private final int count;
+    private final boolean hasBonusNumber;
+
+    Ranking(int prize, int count, boolean hasBonusNumber) {
+        this.prize = prize;
+        this.count = count;
+        this.hasBonusNumber = hasBonusNumber;
+    }
+
+    public static Ranking findRanking(int count, boolean hasBonusNumber) {
+        return Arrays.stream(Ranking.values())
+                .filter(ranking -> checkCountAndBonusNumber(ranking, count, hasBonusNumber))
+                .findAny()
+                .orElse(NONE);
+    }
+
+    private static boolean checkCountAndBonusNumber(Ranking ranking, int count, boolean hasBonusNumber) {
+        if (ranking.count != count) {
+            return false;
+        }
+        if (count == SECOND.count) {
+            return ranking.hasBonusNumber == hasBonusNumber;
+        }
+        return true;
+    }
```

Comment:
```suggestion
    public static Ranking findRanking(int count, boolean hasBonusNumber) {
        if (hasBonusNumber == SECOND.hasBonusNumber && count == SECOND.count) {
            return SECOND;
        }
        return Arrays.stream(Ranking.values())
                .filter(ranking -> ranking != SECOND && ranking.count == count)
                .findAny()
                .orElse(NONE);
    }
```
위처럼 작성할 수도 있을 것 같은데요. 로직이 약~~~간은 복잡하네요. 조금 더 가독성을 높힐 수 있는 방법이 있을까요?